main.js

```diff
@@ -69,4 +117,86 @@ define(function (require, exports, module) {
         // Register nohost server with highest priority
         LiveDevServerManager.registerServer({ create: _getServer }, 9001);
     });
+
+    AppInit.appReady(function (){
+        // When the app is loaded and ready, hide the menus/toolbars
+        HideUI.hide();
+
+        // Once the app has loaded our file,
+        // and we can be confident the editor is open,
+        // get a reference to it and attach our "onchange"
+        // listener to codemirror
+        codeMirror = EditorManager.getActiveEditor()._codeMirror;
+
+        parentWindow.postMessage(JSON.stringify({
+            type: "bramble:change",
+            sourceCode: codeMirror.getValue()
+        }), "*");
+
+        codeMirror.on("change", function(e){
+            parentWindow.postMessage(JSON.stringify({
+                type: "bramble:change",
+                sourceCode: codeMirror.getValue()
+            }), "*");
+        });
+    });
+
+    // Eventually, we'll listen for a message from
+    // thimble containing the make's initial code.
+    // For now, we are defaulting to thimble's starter
+    // make.
+    exports.initExtension = function() {
+        var deferred = new $.Deferred();
+        var data;
+
+        function _getInitialDocument(e) {
+            try {
+                data = e.data || null;
+                data = JSON.parse(data);
+                data = data || {};
+            } catch(err) {
+                // Quick fix: Ignore the 'process-tick' message being sent
+                if(e.data === 'process-tick') {
+                    return;
+                }
+
+                console.error("Parsing message from thimble failed: ", err);
+
+                deferred.reject();
+                return;
+            }
+
+            // Remove the listener after we confirm the event is the
+            // one we're waiting for
+            if (data.type !== "bramble:init") {
+                return;
+            }
+
+            window.removeEventListener("message", _getInitialDocument);
```

Comment:
Combine this with the block above to make it clear they're logically related. Something like: 

```js
if (data.type !== "bramble:init") {
  return;
}
window.Remove....
```